WalletWasabi.Gui/Tabs/WalletManager/RecoverWallets/RecoverWalletViewModel.cs

```diff
@@ -220,16 +220,24 @@ private void ValidateMinGapLimit(IValidationErrors errors)
 			}
 		}
 
-		private void ValidateKeyPath(IValidationErrors errors)
+		private void ValidateAccountKeyPath(IValidationErrors errors)
 		{
 			if (string.IsNullOrWhiteSpace(AccountKeyPath))
 			{
 				errors.Add(ErrorSeverity.Error, "Path is not valid.");
 			}
-			else if (!KeyPath.TryParse(AccountKeyPath, out _))
+			else if (!KeyPath.TryParse(AccountKeyPath, out var keyPath))
 			{
 				errors.Add(ErrorSeverity.Error, "Path is not a valid derivation path.");
 			}
+			else
+			{
+				var accountKeyPath = keyPath.GetAccountKeyPath();
+				if (keyPath.Length != accountKeyPath.Length || accountKeyPath.Length != KeyManager.DefaultAccountKeyPath.Length)
+				{
+					errors.Add(ErrorSeverity.Error, "Path is not a compatible account derivation path.");
+				}
```

Comment:
Nit: we can inverse this if statement
```suggestion
			else if (KeyPath.TryParse(AccountKeyPath, out var keyPath))
			{
				var accountKeyPath = keyPath.GetAccountKeyPath();
				if (keyPath.Length != accountKeyPath.Length || accountKeyPath.Length != KeyManager.DefaultAccountKeyPath.Length)
				{
					errors.Add(ErrorSeverity.Error, "Path is not a compatible account derivation path.");
				}
			}
			else
			{
				errors.Add(ErrorSeverity.Error, "Path is not a valid derivation path.");
```